manage/ships/edit.php

```diff
@@ -0,0 +1,147 @@
+<?php 
+// Display already entered data and allow uses to edit it. Doing similar checks to new
+
+require '../layout/header.php';
+
+use Model\AbstractShip;
+use Service\Container;
+
+$container = new Container($configuration);
+$shipLoader = $container->getShipLoader();
+$teams = AbstractShip::getTeams();
+$errors = [];
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    // set the shipID and instantiate a ship
+    if (isset($_POST['shipId'])) {
+        $id = $_POST['shipId'];
+        $ship = $shipLoader->findOneById($id);
+        if ($ship == null) {
+            echo 'SHIP NOT FOUND';
+            die;
+        }
+    } else {
+        throw new \Exception('NO SHIP WITH THIS ID FOUND');
+        // or could catch and die the error
+    }
+
+    $name = $_POST['name'];
+    if (!empty($_POST['name'])) {
+        $name = $_POST['name'];
+    } else {
+        $errors[] = 'This ship needs a name';
+    }
+    // numeric vals
+    $weaponPower = $_POST['weapon-power'];
+    if (empty($weaponPower)){
+        $weaponPower = 0;
+    } elseif (!is_numeric($weaponPower)) {
+        $errors[] = 'Weapon power must be a number';
+    }
+    // numeric vals
+    $jediFactor = $_POST['jedi-factor'];
+    if (empty($jediFactor)) {
+        $jediFactor = 0;
+    } elseif (!is_numeric($jediFactor)){
+        $errors[] = 'Jedi factor must be a number';
+    }
+    // change to be full health/ MAXHealth
+    // check for empty and numeric values
+    $strength = $_POST['strength'];
+    if (empty($strength)) {
+        $strength = 0;
+    } elseif (!is_numeric($strength)) {
+        $errors[] = 'Strength must be a number';
+    }
+    
+    $team = $_POST['team'];
+    if (empty($team)) {
+        $errors[] = 'All ships must have an allegiance';
+    } elseif (!in_array($team, $teams)) {
+        $errors[] = 'Select a valid team';
+    }
+
+    // check for the ID of the ship 
+
+    if (count($errors) == 0) {
+        $ship->setName($name);
+        $ship->setWeaponPower($weaponPower);
+        $ship->setJediFactor($jediFactor);
+        $ship->setStrength($strength);
+        $ship->setType($team); // may need to change the getType func.
```

Comment:
I think on the edit, you can just disable the type option. Let's say that you can update all the values except type.

This works, but since we have the `Ship` and `RebelShip` classes, I think it makes sense to just not cross their paths.